draft-ietf-lake-traces.md

```diff
@@ -2817,6 +2838,94 @@ OSCORE Master Salt after KeyUpdate (Raw Value) (8 bytes)
 73 ce 79 24 59 40 36 80
 ~~~~~~~~
 
+# Invalid Traces  {#sec-trace-invalid}
+
+This section contains examples of invalid messages, which a compliant implementation must or should reject according to {{I-D.ietf-lake-edhoc}}, {{RFC9053}}, and {{SP-800-56A}}. This is just a small set of examples of different reasons a message might be invalid. The same types of invalidities applies to other fields and messages as well. Implementations should make sure to check for similar types of in invalidities in all EHDOC fields and messages.
+
+Invalid encoding of C_I = 0x0e
+
+~~~~~~~~
+Invalid message_1 (38 bytes)
+03 02 58 20 74 1a 13 d7 ba 04 8f bb 61 5e 94 38 6a a3 b6 1b ea 5b 3d
+8f 65 f3 26 20 b7 49 be e8 d2 78 ef a9 41 0e
+~~~~~~~~
+
+Invalid encoding as array instead of CBOR sequence.
+
+~~~~~~~~
+Invalid message_1 (38 bytes)
+84 03 02 58 20 74 1a 13 d7 ba 04 8f bb 61 5e 94 38 6a a3 b6 1b ea 5b
+3d 8f 65 f3 26 20 b7 49 be e8 d2 78 ef a9 0e
+~~~~~~~~
+
+Invalid 16-bit encoding of METHOD = 3
+
+~~~~~~~~
+Invalid message_1 (39 bytes)
+19 00 03 02 58 20 74 1a 13 d7 ba 04 8f bb 61 5e 94 38 6a a3 b6 1b ea
+5b 3d 8f 65 f3 26 20 b7 49 be e8 d2 78 ef a9 0e
+~~~~~~~~
+
+Invalid type of the third element (G_X).
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 78 20 20 61 69 72 20 73 70 65 65 64 20 6F 66 20 61 20 75 6E 6C
+61 64 65 6E 20 73 77 61 6C 6C 6F 77 20 0e
+~~~~~~~~
+
+Invalid length of third element (G_X). Leading-zeros not preserved.
+
+~~~~~~~~
+Invalid message_1 (36 bytes)
+03 02 58 1f d9 69 77 25 d2 3a 68 8b 12 d1 c7 e0 10 8a 08 c9 f7 1a 85
+a0 9c 20 81 49 76 ab 21 12 22 48 fc 0e
+~~~~~~~~
+
+Invalid x-coordinate in G_X as x {{{≥}}} p
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 58 20 ff ff ff ff 00 00 00 01 00 00 00 00 00 00 00 00 00 00 00
+00 ff ff ff ff ff ff ff ff ff ff ff ff 0e
+~~~~~~~~
+
+Invalid x-coordinate in (G_X) not corresponding to a point on the P-256 curve.
+
+~~~~~~~~
+Invalid message_1 (37 bytes)
+03 02 58 20 a0 4e 73 60 1d f5 44 a7 0b a7 ea 1e 57 03 0f 7d 4b 4e b7
+f6 73 92 4e 58 d5 4c a7 7a 5e 7d 4d 4a 0e
+~~~~~~~~
+
+Invalid number of elements in the CBOR sequence
+
+~~~~~~~~
+Invalid message_2 (46 bytes)
+58 20 41 97 01 d7 f0 0a 26 c2 dc 58 7a 36 dd 75 25 49 f3 37 63 c8 93
+42 2c 8e a0 f9 55 a1 3a 4f f5 d5 4B 98 62 a1 1d e4 2a 95 d7 85 38 6a
+~~~~~~~~
+
+Invalid encoding of ID_CRED_R in PLAINTEXT_2
+
+~~~~~~~~
+Invalid  PLAINTEXT_2 (15 bytes)
+27 a1 04 42 32 10 48 fa 5e fa 2e bf 92 0b f3
+~~~~~~~~
```

Comment:
(This comment was made on a commit, but I add it here too to be sure it's not missed)



What is the invalidity in the new example?

This seems the correct way to encode the credential identifier of type 'kid' with value 0x3210.

Reply:
My understanding of Section 3.5.3.2. is that the CBOR map { 4 : kid_x } is always replaced by the byte string kid_x.

I.e. the correct encoding of this PLAINTEXT_2 is 
27 42 32 10 48 fa 5e fa 2e bf 92 0b f3

Reply:
Ops, you're right :-)

Maybe it's useful if these examples shortly mention what their invalidity is, possibly pointing to the relevant section of -lake-edhoc

Reply:
That is a good idea

Reply:
Done